### PR TITLE
Setup code quality monitoring

### PR DESCRIPTION
**Pull request by ethanschreur**
**What does this PR do?**
- This pull request sets up the code quality monitoring with Codacy

**Description of the task to be completed** 
- Add the repository to be monitored by Codacy
- Add a code quality badge to README.md

**How should this manually be tested**
- None

### CODE REVIEW

**Comment by alexkayabula** [comment]
@ethanschreur is this PR ready for review ?

**Comment by ethanschreur** [comment]
> @ethanschreur is this PR ready for review ?

I think it is ready. I don't believe this PR is dependent on any of the other PRs.

**Comment by alexkayabula** [comment]
> > @ethanschreur is this PR ready for review ?
> 
> I think it is ready. I don't believe this PR is dependent on any of the other PRs.

What do you mean by `I don't believe this PR is dependent on any of the other PRs` ?

**Comment by alexkayabula** [comment]
@ethanschreur the label still shows work in progress. Are you still working on something in this PR ?

**Comment by alexkayabula** [comment]
@ethanschreur the PR shows its `ready for review` but some checks are still running ?

**Comment by ethanschreur** [comment]
Sorry for the confusion. I fixed the header in the readme file after I realized it should be a header and not bolded. 

And by "dependent on any of the other PRs"  I meant that the tasks for this feature are not waiting for any other pull requests to be merged into master before they can be worked on.

**Comment by ethanschreur** [comment]
> @ethanschreur the PR shows its `ready for review` but some checks are still running !!

It shouldn't be ready for review then. My bad.

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
-venv/
\ No newline at end of file
+venv/
+**__pycache__/
```
Comment (alexkayabula) [inline .gitignore]: How is this related to setting up code quality monitoring ?

**Review by alexkayabula — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (ethanschreur) [reply .gitignore]: > How is this related to setting up code quality monitoring ?

I take it this part should go in a different pull request? I'm gonna have to fix that on some other PRs too then.

**Review by alexkayabula — APPROVED** [review]
LGTM. Thank you.
